plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastoreModule.java

```diff
@@ -35,27 +36,35 @@ public HiveMetastoreModule(Optional<HiveMetastore> metastore)
         this.metastore = metastore;
     }
 
+    @SuppressWarnings("Convert2MethodRef")
     @Override
     protected void setup(Binder binder)
     {
         if (metastore.isPresent()) {
             binder.bind(HiveMetastoreFactory.class).annotatedWith(RawHiveMetastoreFactory.class).toInstance(HiveMetastoreFactory.ofInstance(metastore.get()));
         }
         else {
-            bindMetastoreModule("thrift", new ThriftMetastoreModule());
-            bindMetastoreModule("file", new FileMetastoreModule());
-            bindMetastoreModule("glue", new GlueMetastoreModule());
-            bindMetastoreModule("alluxio", new AlluxioMetastoreModule());
+            bindMetastoreModule("thrift", () -> new ThriftMetastoreModule());
+            bindMetastoreModule("file", () -> new FileMetastoreModule());
+            bindMetastoreModule("glue", () -> new GlueMetastoreModule());
+            bindMetastoreModule("alluxio", () -> new AlluxioMetastoreModule());
```

Comment:
Use `Class.forName` for AlluxioMetastoreModule + good comment.
restore the rest.

Reply:
> since it's really only about Alluxio module

Can we make sure it is only about `AlluxioMetastoreModule`. E.g. add a test which scans the class path and checks if only references to Aluxio library are coming from `AlluxioMetastoreModule` (possibly transitively)?

Reply:
We thought about a test like this, but I'm not convinced that it deserves this much effort put into it.